os/kernel/task/task_prctl.c

```diff
@@ -133,14 +135,12 @@ int prctl(int option, ...)
 		int pid = va_arg(ap, int);
 		FAR struct tcb_s *tcb;
 
-		/* Get the TCB associated with the PID (handling the special case of
-		 * pid==0 meaning "this thread")
-		 */
-
-		if (!pid) {
-			tcb = this_task();
-		} else {
+		if (option == PR_SET_NAME_BYPID || option == PR_GET_NAME_BYPID) {
```

Comment:
Need to add `(option == PR_SET_NAME_BYPID)` in line 166

Reply:
Added. Thanks.